source/java/src/org/lucee/extension/cache/eh/EHCacheSupport.java

```diff
@@ -24,15 +24,35 @@
 import lucee.commons.io.cache.CacheEntry;
 import lucee.commons.io.cache.CachePro;
 import lucee.runtime.type.Struct;
+import lucee.runtime.config.Config;
 import net.sf.ehcache.Element;
 import net.sf.ehcache.config.CacheConfiguration;
 
 import org.lucee.extension.cache.CacheSupport;
 import org.lucee.extension.cache.eh.util.TypeUtil;
+import lucee.loader.engine.CFMLEngineFactory;
+import lucee.commons.io.log.Log;
 
 public abstract class EHCacheSupport extends CacheSupport implements Cache {
 	
 	protected boolean isDistributed;
+	protected boolean isSerialized;
+	protected Log logger;
+
+	protected Log getLogger() {
+		return getLogger(null);
+	}
+
+	protected Log getLogger(Config config) {
+		Log logger = (config==null?CFMLEngineFactory.getInstance().getThreadConfig():config).getLog("application");
```

Comment:
why store to a variable and not simply return?

Reply:
In the comment below that line you'll see:

```
// for some reason, setting the application log to "debug" does not always show
// the ehCache output, so when debugging code, we can just manually set the log
// to DEBUG mode to make sure we see the log output
// logger.setLogLevel(logger.LEVEL_DEBUG);
```

I think perhaps this was a bug in Lucee that you recently fixed, but when developing I've had to uncomment the `logger.setLogLevel(logger.LEVEL_DEBUG)` to ensure that the log output is actually written.

If it weren't for that, I would have it as one line.